tests/xharness/Jenkins/Reports/HtmlReportWriter.cs

```diff
@@ -405,8 +405,12 @@ <li>Toggle visibility
 									candidates = (runTest as RunSimulatorTask)?.Candidates;
 								if (candidates != null) {
 									writer.WriteLine ($"Candidate devices:<br />");
-									foreach (var candidate in candidates)
-										writer.WriteLine ($"&nbsp;&nbsp;&nbsp;&nbsp;{candidate.Name} (Version: {candidate.OSVersion})<br />");
+									try {
+										foreach (var candidate in candidates)
+											writer.WriteLine ($"&nbsp;&nbsp;&nbsp;&nbsp;{candidate.Name} (Version: {candidate.OSVersion})<br />");
```

Comment:
Moving the `try catch` inside the `foreach` would list all the _valid_ candidates - not just the first ones before a bad entry.

Not sure if that can happen in real-life ?!? or if it would be useful when debugging such issue...

Reply:
It's the iteration of the `candidates` variable that throws the exception, so if I put the `try catch` inside, it won't catch anything.